docroot/modules/custom/nvli_custom_search/src/Plugin/Block/ResourceTypeTitleBlock.php

```diff
@@ -47,6 +47,9 @@ public function build($resource_type = NULL) {
       '#title' => $title,
       '#resource_id' => $filterID,
       '#resource_link' => NULL,
+      '#cache' => array(
```

Comment:
@piyuesh23 
If you click on any right side resource type then title is cached so it's not changing. Becuase of that I added the cache.